core/src/main/scala/org/ensime/config/richconfig.scala

```diff
@@ -34,17 +34,19 @@ package object richconfig {
         name.startsWith("scala-library") && name.endsWith(".jar")
       }
 
-    def findProject(path: Path): Option[EnsimeProject] = {
+    def findProject(path: Path): Option[EnsimeProjectId] = {
       // should use NIO relations instead of string comparison...
       // needs https://github.com/ensime/ensime-server/issues/1791
       val p = path.toFile.getPath
-      c.projects.find(_.sources.exists(f => p.startsWith(f.getPath)))
+      c.projects collectFirst {
+        case project if project.sources.exists(f => p.startsWith(f.getPath)) => project.id
```

Comment:
since we have `Path` objects, there is no reason to drop down into `String`, we can use `nio` to do a (much more efficient) ancestor check.